libs/constrained-generators/src/Constrained/Base.hs

```diff
@@ -2899,7 +2980,7 @@ preMapFoldSpec :: HasSpec fn a => fn '[a] b -> FoldSpec fn b -> FoldSpec fn a
 preMapFoldSpec _ NoFold = NoFold
 preMapFoldSpec f (FoldSpec g s) = FoldSpec (composeFn g f) s
 
-combineFoldSpec :: MonadGenError m => FoldSpec fn a -> FoldSpec fn a -> m (FoldSpec fn a)
+combineFoldSpec :: FoldSpec fn a -> FoldSpec fn a -> Either [String] (FoldSpec fn a)
```

Comment:
Because one doesn't have to use the run monad functions. Which don't distinguish beween  grnError and fatalError, so this leaves the choice up to the call site.

Reply:
You should be able to just add a `MonadGenError` instance for `Either [String]`

Reply:
Yes and no, because it would fail to distinguish between genError and fatalError. Here it is clear there is only one choice of error, and the context gets to decide.